project-koku/koku
# COST-287 added env variable to disable view caching during development

added CACHED_VIEWS_DISABLED to example env (False by default) and when set to True will not cache views in redis by using django's DummyCache.

Testing:

1. Change CACHED_VIEWS_DISABLED to True in .env file

2. Access the redis container using the following command: `docker exec -it koku_redis bash`

3. Enter the following commands into the redis cli: 
        - `redis-cli FLUSHALL` to reset anything currently cached
        - `redis-cli MONITOR` to see commands processed by the redis server

4. Load any endpoint and watch the redis cli for GET and SET commands, for example: [http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?group_by[org_unit_id]=R_001](http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?group_by[org_unit_id]=R_001) 

5. Ensure none of the GET and SET commands are related to views, some will be related to RBAC but there should not be any related to views, for example you should not see anything like this:
     `"GET" "acct10001::1:views.decorators.cache.cache_header.aws-view.4804a6e5d19a9610d9a2aff378093531.en-us.UTC"`

6. Reload the endpoint and you should not see a GET related to views.
